src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -146,6 +208,14 @@ async Task EnsureLong()
                 {
                     throw new TimeoutException("Timeout while reading from stream");
                 }
+                catch (OperationCanceledException e)
+                {
+                    if (dontBreakOnTimeouts)
+                        throw new TimeoutException("Timeout while reading from stream");
+                    else
+                        throw Connector.Break(new NpgsqlException("Exception while reading from stream",
+                            new TimeoutException("Timeout during reading attempt", e)));
```

Comment:
Not sure the OperationCanceledException should be passed as an inner exception here - it conveys very little (no?) additional information, and the cancellation token is an internal implementation detail of how the timeout is implemented...

If we do want to surface this as an inner exception, we should do so also for the TimeoutException being thrown above (under `dontBreakOnTimeouts`)